ortools/linear_solver/xpress_interface_test.cc

```diff
@@ -1253,6 +1258,144 @@ TEST(XpressInterface, setInt64Control) {
+            solver.solver_infinity());
+}
+
+TEST_F(XpressFixtureMIP, mixedRhsTypes) {
+  MPVariable* var1 = solver.MakeVar(-1e20, 1e20, false, "var1");
+  MPVariable* var2 = solver.MakeVar(-2000, 2000, false, "var2");
+  MPVariable* var3 = solver.MakeVar(0, 1e20, false, "var3");
+  MPVariable* var4 = solver.MakeVar(-6, 6, false, "var4");
+  MPVariable* var5 = solver.MakeVar(0, 1e20, false, "var5");
+  MPVariable* var6 = solver.MakeVar(-1e20, 1e20, false, "var6");
+  MPVariable* var7 = solver.MakeVar(0, 1, true, "var7");
+  MPVariable* var8 = solver.MakeVar(0, 1, true, "var8");
+  MPVariable* var9 = solver.MakeVar(0, 1, false, "var9");
+
+  MPConstraint* const1 = solver.MakeRowConstraint(932, 932, "const1");
+  const1->SetCoefficient(var1, 1);
+  const1->SetCoefficient(var2, 0.7);
+  const1->SetCoefficient(var4, 6);
+  MPConstraint* const2 = solver.MakeRowConstraint(0, 1e20, "const2");
+  const2->SetCoefficient(var3, 1);
+  const2->SetCoefficient(var2, -1);
+  MPConstraint* const3 = solver.MakeRowConstraint(0, 1e20, "const3");
+  const3->SetCoefficient(var3, 1);
+  const3->SetCoefficient(var2, 1);
+  MPConstraint* const4 = solver.MakeRowConstraint(0, 1e20, "const4");
+  const4->SetCoefficient(var4, -1);
+  const4->SetCoefficient(var5, 1);
+  MPConstraint* const5 = solver.MakeRowConstraint(0, 1e20, "const5");
+  const5->SetCoefficient(var4, 1);
+  const5->SetCoefficient(var5, 1);
+  MPConstraint* const6 = solver.MakeRowConstraint(-1e20, 300, "const6");
+  const6->SetCoefficient(var6, 1);
+  const6->SetCoefficient(var1, -1);
+  MPConstraint* const7 = solver.MakeRowConstraint(-1e20, 300, "const7");
+  const7->SetCoefficient(var6, 1);
+  const7->SetCoefficient(var1, 1);
+  MPConstraint* const8 = solver.MakeRowConstraint(-1e20, 0.0001, "const8");
+  const8->SetCoefficient(var3, 1);
+  const8->SetCoefficient(var7, -4000);
+  MPConstraint* const9 = solver.MakeRowConstraint(-1e20, 0.0001, "const9");
+  const9->SetCoefficient(var5, 1);
+  const9->SetCoefficient(var8, -12);
+  MPConstraint* const10 = solver.MakeRowConstraint(0, 1, "const10");
+  const10->SetCoefficient(var7, 1);
+  const10->SetCoefficient(var8, 1);
+  MPConstraint* const11 = solver.MakeRowConstraint(0, 2e9, "const11");
+  const11->SetCoefficient(var9, 1);
+  MPConstraint* const12 = solver.MakeRowConstraint(0, 1e20, "const12");
+  const12->SetCoefficient(var9, 1);
+  const12->SetCoefficient(var7, -1);
+  MPConstraint* const13 = solver.MakeRowConstraint(0, 1e20, "const13");
+  const13->SetCoefficient(var9, 1);
+  const13->SetCoefficient(var8, -1);
+
+  solver.MutableObjective()->SetMinimization();
+  solver.MutableObjective()->SetCoefficient(var3, 0.001);
+  solver.MutableObjective()->SetCoefficient(var5, 0.01);
+  solver.MutableObjective()->SetCoefficient(var6, -1);
+
+  // disable formatting to keep the expected MPS readable
+  // clang-format off
+  std::string expectedMps = std::string("") +
+                            "NAME          newProb" + "\n" +
```

Comment:
Use multiline string ? Example

```cpp
const char* p = R"(Hello
world)";

#include <iostream>
int main() {
  std::cout << p << std::endl;
}
```

Reply:
I think this doesn't work with trailing spaces (e.g. " L  const9  " + "\n"), since they can be removed by auto-formatting